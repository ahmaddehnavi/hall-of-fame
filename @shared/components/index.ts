export * from './layout/Col'
export * from './layout/Row'
export * from './screen/Screen'
export * from './backhandler/MultiBackHandler'
export * from './touchable/Touchable'
export * from './button/Button'
export * from './list/FlatList'
export * from './empty/EmptyView'
export * from './loading/LoadingView'
export * from './error/ErrorView'
export * from './async-view/AsyncView'