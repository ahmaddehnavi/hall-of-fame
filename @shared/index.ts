export * from './services'
export * from './utils'
export * from './components'
export * from './resource/Resource'
