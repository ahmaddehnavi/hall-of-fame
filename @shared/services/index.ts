export * from './base'
export * from './navigation'
export * from './theme'

