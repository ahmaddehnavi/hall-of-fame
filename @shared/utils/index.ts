export * from './di/DIUtil'
export * from './async/AsyncUtil'