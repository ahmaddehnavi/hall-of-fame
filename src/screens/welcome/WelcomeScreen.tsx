import {
    DIInject,
    INavigationService,
    InjectedNavigationServiceProps,
    InjectedThemeServiceProps,
    MultiBackHandler,
    NavigationService,
    Screen,
    ThemeService
} from '@shared';
import autobind from 'autobind-decorator';
import {observer} from 'mobx-react';
import React from 'react';
import {BackHandler, StyleSheet} from 'react-native';
import {InjectedWelcomeStoreProps, WelcomeStore} from '../../stores/WelcomeStore';
import {SoundUtil} from '../../utils/SoundUtil';
import Assets from '../../assets/Assets';
import {FameListScreen} from '../fame-list/FameListScreen';
import {WelcomeComponent} from './WelcomeComponent';

type WelcomeScreenProps =
    InjectedNavigationServiceProps &
    InjectedThemeServiceProps &
    InjectedWelcomeStoreProps


@DIInject(NavigationService.NAME, ThemeService.NAME, WelcomeStore.NAME)
@observer
export class WelcomeScreen extends React.Component<WelcomeScreenProps> {
    static readonly ROUTE_NAME = 'WelcomeScreen';

    static start(nav: INavigationService) {
        nav.navigate(this.ROUTE_NAME);
    }

    static resetTo(nav: INavigationService) {
        nav.reset(this.ROUTE_NAME);
    }

    componentDidMount() {
        this.props.$welcomeStore.startAnimationChanger();
    }

    componentWillUnmount(): void {
        this.props.$welcomeStore.stopAnimationChanger();
    }

    render() {
        let store = this.props.$welcomeStore;
        let activeAnimation = store.activeAnimation;
        return (
            <Screen style={styles.container}>

                <WelcomeComponent
                    onRandomisePress={store.randomise}
                    oSavePress={store.save}
                    onNumberChanged={store.updateNumber}
                    inputValue={String(store.numberValue)}
                    animation={activeAnimation.animationName}
                    image={activeAnimation.image}
                    $theme={this.props.$theme}
                />

                <MultiBackHandler
                    timeout={500}
                    maxCount={2}
                    onPress={this.handleBackPress}/>
            </Screen>
        )
    }

    @autobind
    handleBackPress(count: number) {
        if (count === 1) {
            FameListScreen.start(this.props.$navigation)
        } else if (count === 2) {
            SoundUtil.play(Assets.sounds.test)
                .then(success => {
                    BackHandler.exitApp();
                })
        }
    }
}


const styles = StyleSheet.create({
    container: {
        alignItems: 'stretch',
        justifyContent: 'center'
    },
});