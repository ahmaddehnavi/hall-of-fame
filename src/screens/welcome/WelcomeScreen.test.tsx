import {NavigationService, ThemeService} from '@shared';
import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {WelcomeScreenStore} from './WelcomeScreen.store';
import {WelcomeScreen} from './WelcomeScreen';

it('renders correctly', () => {
    let tree = renderer.create(
        <WelcomeScreen
            $navigation={new NavigationService()}
            $welcomeStore={new WelcomeScreenStore()}
            $theme={new ThemeService()}
        />
    );
    expect(tree).toMatchSnapshot();
});

