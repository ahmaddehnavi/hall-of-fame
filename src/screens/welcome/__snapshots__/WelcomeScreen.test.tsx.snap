// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "stretch",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "paddingHorizontal": 37.5,
          "paddingVertical": 66.7,
        }
      }
    >
      <View
        onResponderRelease={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="auto"
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "rgba(0, 0, 0, 0.38)",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 8,
              "paddingTop": 32,
            }
          }
        >
          <View
            style={
              Object {
                "position": "absolute",
                "top": 16,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "rgba(0, 0, 0, 0.38)",
                  "fontSize": 12,
                }
              }
            >
              Enter a number
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="sentences"
              disableFullscreenUI={true}
              editable={true}
              keyboardType="numeric"
              maxLength={1}
              onBlur={[Function]}
              onChange={[Function]}
              onChangeText={[Function]}
              onContentSizeChange={[Function]}
              onFocus={[Function]}
              selectionColor="rgb(0, 145, 234)"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "margin": 0,
                    "padding": 0,
                    "top": 2,
                  },
                  Object {
                    "color": "rgba(0, 0, 0, .87)",
                    "fontSize": 16,
                    "height": 24,
                    "textAlign": "left",
                  },
                  undefined,
                ]
              }
              underlineColorAndroid="transparent"
              value="1"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "height": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgb(213, 0, 0)",
                    "fontSize": 0,
                    "opacity": 0,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "bottom": 0,
                  "left": 0,
                  "paddingVertical": 4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(0, 0, 0, .38)",
                    "fontSize": 12,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 2,
                  },
                  Object {
                    "backgroundColor": undefined,
                  },
                  Object {
                    "flex": 1,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "paddingHorizontal": 12,
                      "paddingVertical": 8,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": undefined,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Save
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 16,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 2,
                  },
                  Object {
                    "borderColor": undefined,
                    "borderWidth": 1,
                  },
                  Object {
                    "flex": 1,
                  },
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "paddingHorizontal": 12,
                      "paddingVertical": 8,
                      "textAlign": "center",
                      "textAlignVertical": "center",
                    },
                    Object {
                      "color": undefined,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Randomise
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
